// @buffetjs/core/dist/components/ListHeader/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _Button = _interopRequireDefault(require("../Button"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ListHeader(_ref) {
  var button = _ref.button,
      title = _ref.title,
      subtitle = _ref.subtitle;
  return _react["default"].createElement(_styles.ListHeader, null, button && _react["default"].createElement(_Button["default"], button), _react["default"].createElement(_styles.ListTitle, null, title), _react["default"].createElement(_styles.ListSubtitle, null, subtitle));
}

ListHeader.defaultProps = {
  button: null,
  title: null,
  subtitle: null
};
ListHeader.propTypes = {
  button: _propTypes["default"].shape({
    color: _propTypes["default"].string,
    icon: _propTypes["default"].bool,
    type: _propTypes["default"].string
  }),
  subtitle: _propTypes["default"].string,
  title: _propTypes["default"].string
};
var _default = ListHeader;
exports["default"] = _default;
