// @buffetjs/core/dist/components/Toggle/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _Label = _interopRequireDefault(require("../Label"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Toggle(_ref) {
  var id = _ref.id,
      className = _ref.className,
      name = _ref.name,
      _onChange = _ref.onChange,
      value = _ref.value;
  return _react["default"].createElement(_styles.ToggleWrapper, {
    className: className
  }, _react["default"].createElement(_Label["default"], {
    htmlFor: id || name
  }, _react["default"].createElement(_styles.Toggle, {
    checked: value,
    id: id || name,
    name: id || name,
    onChange: function onChange(e) {
      _onChange({
        target: {
          name: name,
          value: e.target.checked
        }
      });
    }
  }), _react["default"].createElement("span", null, "OFF"), _react["default"].createElement("span", null, "ON")));
}

Toggle.defaultProps = {
  className: null,
  id: null,
  onChange: function onChange() {},
  value: false
};
Toggle.propTypes = {
  className: _propTypes["default"].string,
  id: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  onChange: _propTypes["default"].func,
  value: _propTypes["default"].bool
};
var _default = Toggle;
exports["default"] = _default;
