// @buffetjs/core/dist/components/Button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodash = require("lodash");

var _styles = require("@buffetjs/styles");

var _icons = require("@buffetjs/icons");

var _PrefixIcon = _interopRequireDefault(require("../PrefixIcon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Button(props) {
  var content = props.label ? props.label : props.children;
  var disabled = props.disabled;
  var icon = props.icon === true ? _react["default"].createElement(_icons.Plus, {
    height: "11px",
    width: "11px"
  }) : _react["default"].createElement(_PrefixIcon["default"], {
    icon: props.icon
  });
  return _react["default"].createElement(_styles.Button, _extends({}, (0, _lodash.omit)(props, ['icon']), {
    disabled: disabled
  }), icon, content);
}

Button.defaultProps = {
  children: null,
  color: 'primary',
  disabled: false,
  icon: false,
  label: null,
  type: 'button'
};
Button.propTypes = {
  children: _propTypes["default"].node,
  color: _propTypes["default"].string,
  disabled: _propTypes["default"].bool,
  icon: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].bool]),
  label: _propTypes["default"].string,
  type: _propTypes["default"].oneOf(['submit', 'reset', 'button', 'menu'])
};
var _default = Button;
exports["default"] = _default;
