// @buffetjs/core/dist/components/DatePicker/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _moment = _interopRequireDefault(require("moment"));

var _reactMomentProptypes = _interopRequireDefault(require("react-moment-proptypes"));

require("react-dates/initialize");

var _reactDates = require("react-dates");

var _freeRegularSvgIcons = require("@fortawesome/free-regular-svg-icons");

var _reactFontawesome = require("@fortawesome/react-fontawesome");

var _styles = require("@buffetjs/styles");

var _InputText = _interopRequireDefault(require("../InputText"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var DatePicker = function (_React$PureComponent) {
  _inherits(DatePicker, _React$PureComponent);

  function DatePicker() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, DatePicker);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(DatePicker)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.state = {
      date: null,
      displayedDate: '',
      isFocused: false,
      visible: false
    };
    _this.timer = null;

    _this.handleChange = function (_ref) {
      var target = _ref.target;

      _this.setState({
        displayedDate: target.value
      });

      clearTimeout(_this.timer);
      _this.timer = setTimeout(function () {
        _this.handleDateChange((0, _moment["default"])(target.value));
      }, _this.props.wait);
    };

    _this.handleDateChange = function (date) {
      var _this$props = _this.props,
          name = _this$props.name,
          onChange = _this$props.onChange;

      if ((0, _moment["default"])(date).isValid()) {
        _this.setState({
          date: date,
          displayedDate: date.format('MM/DD/YYYY'),
          visible: true
        }, function () {
          return onChange({
            target: {
              name: name,
              type: 'date',
              value: date
            }
          });
        });
      }
    };

    _this.handleFocusChange = function (_ref2) {
      var focused = _ref2.focused;

      _this.setState({
        isFocused: focused
      });
    };

    _this.handleOutsideClick = function () {
      _this.setState({
        visible: false
      });
    };

    _this.handleTabClick = function (_ref3) {
      var keyCode = _ref3.keyCode,
          which = _ref3.which;
      var code = keyCode || which;

      if (code === 9) {
        _this.handleOutsideClick();
      }
    };

    _this.showDatepicker = function () {
      _this.setState({
        visible: true
      });
    };

    return _this;
  }

  _createClass(DatePicker, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this$props2 = this.props,
          value = _this$props2.value,
          withDefaultValue = _this$props2.withDefaultValue;
      var date = null;
      var displayedDate = '';

      if (withDefaultValue && !value) {
        date = (0, _moment["default"])();
        displayedDate = date.format('MM/DD/YYYY');
      }

      if (!!value && (0, _moment["default"])(value).isValid()) {
        date = value._isAMomentObject === true ? value : (0, _moment["default"])(value);
        displayedDate = date.format('MM/DD/YYYY');
      }

      this.setState({
        date: date,
        displayedDate: displayedDate
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props3 = this.props,
          className = _this$props3.className,
          disabled = _this$props3.disabled,
          displayFormat = _this$props3.displayFormat,
          id = _this$props3.id,
          name = _this$props3.name,
          readOnly = _this$props3.readOnly,
          tabIndex = _this$props3.tabIndex;
      var _this$state = this.state,
          date = _this$state.date,
          displayedDate = _this$state.displayedDate,
          isFocused = _this$state.isFocused,
          visible = _this$state.visible;
      var dateValue = date ? (0, _moment["default"])(date).format(displayFormat) : '';

      if (visible) {
        dateValue = displayedDate;
      }

      return _react["default"].createElement(_styles.DatePicker, {
        className: "".concat(className, " ").concat(visible && 'is-open')
      }, _react["default"].createElement("div", null, _react["default"].createElement(_InputText["default"], {
        disabled: disabled,
        type: "text",
        name: "start_date",
        id: id || name,
        value: dateValue,
        readOnly: readOnly,
        onChange: this.handleChange,
        icon: _react["default"].createElement(_reactFontawesome.FontAwesomeIcon, {
          icon: _freeRegularSvgIcons.faCalendarAlt
        }),
        onClick: this.showDatepicker,
        onFocus: this.showDatepicker,
        onKeyDown: this.handleTabClick,
        tabIndex: tabIndex
      })), visible && _react["default"].createElement(_reactDates.DayPickerSingleDateController, {
        date: date,
        focused: isFocused,
        numberOfMonths: 1,
        keepOpenOnDateSelect: true,
        onFocusChange: this.handleFocusChange,
        onDateChange: this.handleDateChange,
        onOutsideClick: this.handleOutsideClick
      }));
    }
  }]);

  return DatePicker;
}(_react["default"].PureComponent);

DatePicker.defaultProps = {
  className: null,
  disabled: false,
  displayFormat: 'MMMM DD, YY',
  id: 'date',
  onChange: function onChange() {},
  readOnly: false,
  tabIndex: '0',
  value: null,
  wait: 600,
  withDefaultValue: false
};
DatePicker.propTypes = {
  className: _propTypes["default"].string,
  disabled: _propTypes["default"].bool,
  displayFormat: _propTypes["default"].string,
  id: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  onChange: _propTypes["default"].func,
  readOnly: _propTypes["default"].bool,
  tabIndex: _propTypes["default"].string,
  value: _propTypes["default"].oneOfType([_reactMomentProptypes["default"].momentObj, _propTypes["default"].string]),
  wait: _propTypes["default"].number,
  withDefaultValue: _propTypes["default"].bool
};
var _default = DatePicker;
exports["default"] = _default;
