// @buffetjs/custom/dist/components/List/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@buffetjs/core");

var _styles = require("@buffetjs/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function List(_ref) {
  var title = _ref.title,
      subtitle = _ref.subtitle,
      button = _ref.button,
      items = _ref.items,
      customRowComponent = _ref.customRowComponent;
  return _react["default"].createElement(_styles.Card, null, _react["default"].createElement(_core.ListHeader, {
    title: title,
    subtitle: subtitle,
    button: button
  }), _react["default"].createElement(_core.List, {
    items: items,
    customRowComponent: customRowComponent
  }));
}

List.defaultProps = {
  button: null,
  customRowComponent: null,
  items: [],
  title: null,
  subtitle: null
};
List.propTypes = {
  button: _propTypes["default"].shape({
    color: _propTypes["default"].string,
    icon: _propTypes["default"].bool,
    type: _propTypes["default"].string
  }),
  customRowComponent: _propTypes["default"].func,
  items: _propTypes["default"].instanceOf(Array),
  subtitle: _propTypes["default"].string,
  title: _propTypes["default"].string
};
var _default = List;
exports["default"] = _default;
